lib/client.js

```diff
@@ -112,16 +112,16 @@ Client.prototype.send = function() {
   // Respect IRC line length limit of 512 chars
   // https://tools.ietf.org/html/rfc2812#section-2.3
   if (message.length > 512) {
-    var parts   = message.split(':');
-    var command = parts[1];
-    var payload = parts[2];
+    var payload_marker  = message.indexOf(' :');
+    var command = message.substr(0, payload_marker);
+    var payload = message.substr(payload_marker + 2);
 
     var chunks = payload.split(' ');
```

Comment:
Done. Had to adapt the test a bit for it to test both cases - the messages list describes a list of irc messages (that is, up to 512 chars), which were previously joined by spaces, but only the original test case needed to join words by spaces. Now it joins by the empty string and the first two messages have a space added at the end, which is trimmed when comparing, and the other two messages are joined as a 1024 char long word of 'a' with no spaces.